Python 3.9 removed old isAlive() function which is long replaced by is_alive():
https://github.com/python/cpython/commit/44046fe4fc7f00a6eb855b33e6a3f953cf5233a5

Upstream already fixed this in newer versions of cheroot.

--- cheroot-6.3.2/cheroot/workers/threadpool.py
+++ cheroot-6.3.2/cheroot/workers/threadpool.py
@@ -206,7 +206,7 @@ class ThreadPool:
         # Grow/shrink the pool if necessary.
         # Remove any dead threads from our list
         for t in self._threads:
-            if not t.isAlive():
+            if not t.is_alive():
                 self._threads.remove(t)
                 amount -= 1
 
@@ -239,7 +239,7 @@ class ThreadPool:
             endtime = time.time() + timeout
         while self._threads:
             worker = self._threads.pop()
-            if worker is not current and worker.isAlive():
+            if worker is not current and worker.is_alive():
                 try:
                     if timeout is None or timeout < 0:
                         worker.join()
@@ -247,7 +247,7 @@ class ThreadPool:
                         remaining_time = endtime - time.time()
                         if remaining_time > 0:
                             worker.join(remaining_time)
-                        if worker.isAlive():
+                        if worker.is_alive():
                             # We exhausted the timeout.
                             # Forcibly shut down the socket.
                             c = worker.conn
